enacts/config-zmd.yaml

```diff
@@ -124,3 +124,32 @@ onset:
             description: The map shows the probability of the onset date to be past a certain date,
                 through all the years of available data.
                 Dates are expressed in days since an Early Start date.
+        length_mean:
+            menu_label: How long is the rainy season?
+            description: The map shows the average length (in days) of the rainy season,
+                defined as the period between onset and cessation dates,
+                over all years of available data.
+            map_max: 180
+        length_stddev:
+            menu_label: How uncertain is the length of the rainy season?
+            description: The map shows the standard deviation from the average length of the season,
+                over all years of available data.
+            map_max: 40
+        length_pe:
+            menu_label: How likely is the rainy season lasting less than...
+            description: The map shows the probability of the rainy season
+                to last less than a certain number of days through all the years of available data.
+        #total_mean:
+        #    menu_label: How much rain falls in the rainy season?
+        #    description: The map shows the average precipitation (in mm)  during the rainy season,
+        #        defined as the period between onset and cessation dates,
+        #        over all years of available data.
+        #total_stddev:
+        #    menu_label: How uncertain is the amount of rain in the rainy season?
+        #    description: The map shows the standard deviation from the average precipitation in the season,
+        #        over all years of available data.
+        #total_pe:
+        #    menu_label: How likely is it raining less than...
+        #    description: The map shows the probability of precipitation in the rainy season
+        #        being less than a certain amount through all the years of available data.
+
```

Comment:
Generally speaking, I think this touches the conflicting needs we have to, on one hand, allow others to configure to what they like/want; and on the other hand to allow ourselves to easily reproduce the same maproom in different instances. It's conflicting because our needs call for homogenization whereas our partners needs call for specificity.

I am certain that folks who can understand the math being done but have no clue or capacity to edit it, could have however different views on how to describe and write in words about it. Besides, with respect to capacity building, it's probably fair to distinguish folks who would not be able ever to edit code but could edit text; and folks who could do both. For all those reasons I think there is value to have text/description in the config file.

Then this raises the question on how someone who has the capacity to change code would go about customizing their Maprooms. If they want to change some of the math, then it's outside the config file and they diverge from our homogeneous case that works for all (who we are the only ones who care about). A change in the math would be fair to call for a new Maproom altogether, which can also be fair, with a risk of high inflation of Maprooms that do more or less the same thing...

So I am not sure how we want to go about that, but I think offering the possibility to customize text would satisfy a great number of potential Maproom contributors who are not the programmers/coders.

All this being said, this could still be set up as default with possibility to change through config files. How would you go about that? There are loops and callbacks that rely on what's in map_text to pull out the description. Should it be set to null in the config file; and at the beginning of layout, start by checking if null and if so assign a default text?

Reply:
OK. The `CONFIG` environment variable is already treated as a semicolon-separated list of files, where  files that come later in the list override values given in earlier files. Let's have a `config-defaults.yaml` that gets included automatically as the first element of the list, and is included in the package rather than provided separately like the site-specific config file(s). (It's currently a meaningless distinction,because even the site-specific stuff is here, but I'm imagining that in the future site-specific configuration will move somewhere else.)

Reply:
ok, so to make sure I understand the plan, we would have: i) one config.yaml with the configuration related to the environment where the app is run; ii) one config-defaults.yaml with configuration that don't have to be site specific for the app to run/make sense; and iii) a series of config-site.yaml with configuration of site-specific elements.
Both -defaults and -site would be set with app-wise sections.

Reply:
Yes, but config-defaults.yaml should be built-in to the application, not something the operator needs to provide on the command line.

As for the app-wise sections, I'll say yes for now, but eventually we might want to break up the default config, because that's probably going to become huge.